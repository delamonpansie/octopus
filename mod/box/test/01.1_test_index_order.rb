#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].enabled = 1
object_space[0].index[0] = { type = "HASH"
			     unique = 1
			     key_field[1] = { fieldno = 1, type = "STR" }
}
object_space[0].index[1] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "NUM" }
}
object_space[0].index[2] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "SNUM32" }
}
object_space[0].index[3] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "UNUM32" }
}
object_space[0].index[4] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "NUM32", sort_order = "DESC" }
}
object_space[0].index[5] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "SNUM32", sort_order = "DESC" }
}
object_space[0].index[6] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "UNUM32", sort_order = "DESC" }
}


object_space[1].enabled = 1
object_space[1].index[0] = { type = "HASH"
			     unique = 1
			     key_field[0] = { fieldno = 1, type = "STR" }
}
object_space[1].index[1] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "NUM64" }
}
object_space[1].index[2] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "SNUM64" }
}
object_space[1].index[3] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "UNUM64" }
}
object_space[1].index[4] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "NUM64", sort_order = "DESC" }
}
object_space[1].index[5] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "SNUM64", sort_order = "DESC" }
}
object_space[1].index[6] = { type = "TREE"
			     unique = 0
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "UNUM64", sort_order = "DESC" }
}


object_space[2].enabled = 1
object_space[2].index[0] = { type = "HASH"
			     unique = 1
			     key_field[0] = { fieldno = 0, type = "STR" }
}
object_space[2].index[1] = { type = "TREE"
			     unique = 1
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "STR" }
}
object_space[2].index[2] = { type = "TREE"
			     unique = 1
                             key_field[0] = { fieldno = 0, type = "STR" }
			     key_field[1] = { fieldno = 1, type = "STR", sort_order = "DESC" }
}


EOD
  end
end

Env.connect_eval do
  ping

  self.object_space = 0
  (-10..10).each do |v|
    insert ["a", v]
  end
  sig = [nil, 'L', 'l', 'L', 'L', 'l', 'L']
  (1..6).each do |i|
    s = select(["a"], :index => i)
    puts s.map {|el| el[1].unpack(sig[i])}.join(", ")
    puts
  end

  self.object_space = 1
  (-10..10).each do |v|
    v = [v].pack('q')
    insert ["a", v]
  end
  sig = [nil, 'Q', 'q', 'Q', 'Q', 'q', 'Q']
  (1..6).each do |i|
    s = select(["a"], :index => i)
    puts s.map {|el| el[1].unpack(sig[i])}.join(", ")
    puts
  end


end
