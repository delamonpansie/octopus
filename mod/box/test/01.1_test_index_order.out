# box.ping()
:pong

# box.object_space=(0)
0

# box.insert(["a", -10])
1

# box.insert(["a", -9])
1

# box.insert(["a", -8])
1

# box.insert(["a", -7])
1

# box.insert(["a", -6])
1

# box.insert(["a", -5])
1

# box.insert(["a", -4])
1

# box.insert(["a", -3])
1

# box.insert(["a", -2])
1

# box.insert(["a", -1])
1

# box.insert(["a", 0])
1

# box.insert(["a", 1])
1

# box.insert(["a", 2])
1

# box.insert(["a", 3])
1

# box.insert(["a", 4])
1

# box.insert(["a", 5])
1

# box.insert(["a", 6])
1

# box.insert(["a", 7])
1

# box.insert(["a", 8])
1

# box.insert(["a", 9])
1

# box.insert(["a", 10])
1

# box.select(["a"], {:index=>1})
[["a", "\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00"], ["a", "\x02\x00\x00\x00"], ["a", "\x03\x00\x00\x00"], ["a", "\x04\x00\x00\x00"], ["a", "\x05\x00\x00\x00"], ["a", "\x06\x00\x00\x00"], ["a", "\a\x00\x00\x00"], ["a", "\b\x00\x00\x00"], ["a", "\t\x00\x00\x00"], ["a", "\n\x00\x00\x00"], ["a", "\xF6\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF"], ["a", "\xFF\xFF\xFF\xFF"]]

0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 4294967286, 4294967287, 4294967288, 4294967289, 4294967290, 4294967291, 4294967292, 4294967293, 4294967294, 4294967295

# box.select(["a"], {:index=>2})
[["a", "\xF6\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF"], ["a", "\xFF\xFF\xFF\xFF"], ["a", "\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00"], ["a", "\x02\x00\x00\x00"], ["a", "\x03\x00\x00\x00"], ["a", "\x04\x00\x00\x00"], ["a", "\x05\x00\x00\x00"], ["a", "\x06\x00\x00\x00"], ["a", "\a\x00\x00\x00"], ["a", "\b\x00\x00\x00"], ["a", "\t\x00\x00\x00"], ["a", "\n\x00\x00\x00"]]

-10, -9, -8, -7, -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10

# box.select(["a"], {:index=>3})
[["a", "\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00"], ["a", "\x02\x00\x00\x00"], ["a", "\x03\x00\x00\x00"], ["a", "\x04\x00\x00\x00"], ["a", "\x05\x00\x00\x00"], ["a", "\x06\x00\x00\x00"], ["a", "\a\x00\x00\x00"], ["a", "\b\x00\x00\x00"], ["a", "\t\x00\x00\x00"], ["a", "\n\x00\x00\x00"], ["a", "\xF6\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF"], ["a", "\xFF\xFF\xFF\xFF"]]

0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 4294967286, 4294967287, 4294967288, 4294967289, 4294967290, 4294967291, 4294967292, 4294967293, 4294967294, 4294967295

# box.select(["a"], {:index=>4})
[["a", "\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF"], ["a", "\xF6\xFF\xFF\xFF"], ["a", "\n\x00\x00\x00"], ["a", "\t\x00\x00\x00"], ["a", "\b\x00\x00\x00"], ["a", "\a\x00\x00\x00"], ["a", "\x06\x00\x00\x00"], ["a", "\x05\x00\x00\x00"], ["a", "\x04\x00\x00\x00"], ["a", "\x03\x00\x00\x00"], ["a", "\x02\x00\x00\x00"], ["a", "\x01\x00\x00\x00"], ["a", "\x00\x00\x00\x00"]]

4294967295, 4294967294, 4294967293, 4294967292, 4294967291, 4294967290, 4294967289, 4294967288, 4294967287, 4294967286, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0

# box.select(["a"], {:index=>5})
[["a", "\n\x00\x00\x00"], ["a", "\t\x00\x00\x00"], ["a", "\b\x00\x00\x00"], ["a", "\a\x00\x00\x00"], ["a", "\x06\x00\x00\x00"], ["a", "\x05\x00\x00\x00"], ["a", "\x04\x00\x00\x00"], ["a", "\x03\x00\x00\x00"], ["a", "\x02\x00\x00\x00"], ["a", "\x01\x00\x00\x00"], ["a", "\x00\x00\x00\x00"], ["a", "\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF"], ["a", "\xF6\xFF\xFF\xFF"]]

10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, -2, -3, -4, -5, -6, -7, -8, -9, -10

# box.select(["a"], {:index=>6})
[["a", "\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF"], ["a", "\xF6\xFF\xFF\xFF"], ["a", "\n\x00\x00\x00"], ["a", "\t\x00\x00\x00"], ["a", "\b\x00\x00\x00"], ["a", "\a\x00\x00\x00"], ["a", "\x06\x00\x00\x00"], ["a", "\x05\x00\x00\x00"], ["a", "\x04\x00\x00\x00"], ["a", "\x03\x00\x00\x00"], ["a", "\x02\x00\x00\x00"], ["a", "\x01\x00\x00\x00"], ["a", "\x00\x00\x00\x00"]]

4294967295, 4294967294, 4294967293, 4294967292, 4294967291, 4294967290, 4294967289, 4294967288, 4294967287, 4294967286, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0

# box.object_space=(1)
1

# box.insert(["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"])
1

# box.insert(["a", "\x00\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\x01\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\x02\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\x03\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\x04\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\x05\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\x06\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\a\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\b\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\t\x00\x00\x00\x00\x00\x00\x00"])
1

# box.insert(["a", "\n\x00\x00\x00\x00\x00\x00\x00"])
1

# box.select(["a"], {:index=>1})
[["a", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x02\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x03\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x04\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x05\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x06\x00\x00\x00\x00\x00\x00\x00"], ["a", "\a\x00\x00\x00\x00\x00\x00\x00"], ["a", "\b\x00\x00\x00\x00\x00\x00\x00"], ["a", "\t\x00\x00\x00\x00\x00\x00\x00"], ["a", "\n\x00\x00\x00\x00\x00\x00\x00"], ["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"]]

0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 18446744073709551606, 18446744073709551607, 18446744073709551608, 18446744073709551609, 18446744073709551610, 18446744073709551611, 18446744073709551612, 18446744073709551613, 18446744073709551614, 18446744073709551615

# box.select(["a"], {:index=>2})
[["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x02\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x03\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x04\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x05\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x06\x00\x00\x00\x00\x00\x00\x00"], ["a", "\a\x00\x00\x00\x00\x00\x00\x00"], ["a", "\b\x00\x00\x00\x00\x00\x00\x00"], ["a", "\t\x00\x00\x00\x00\x00\x00\x00"], ["a", "\n\x00\x00\x00\x00\x00\x00\x00"]]

-10, -9, -8, -7, -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10

# box.select(["a"], {:index=>3})
[["a", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x02\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x03\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x04\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x05\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x06\x00\x00\x00\x00\x00\x00\x00"], ["a", "\a\x00\x00\x00\x00\x00\x00\x00"], ["a", "\b\x00\x00\x00\x00\x00\x00\x00"], ["a", "\t\x00\x00\x00\x00\x00\x00\x00"], ["a", "\n\x00\x00\x00\x00\x00\x00\x00"], ["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"]]

0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 18446744073709551606, 18446744073709551607, 18446744073709551608, 18446744073709551609, 18446744073709551610, 18446744073709551611, 18446744073709551612, 18446744073709551613, 18446744073709551614, 18446744073709551615

# box.select(["a"], {:index=>4})
[["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\n\x00\x00\x00\x00\x00\x00\x00"], ["a", "\t\x00\x00\x00\x00\x00\x00\x00"], ["a", "\b\x00\x00\x00\x00\x00\x00\x00"], ["a", "\a\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x06\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x05\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x04\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x03\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x02\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x00\x00\x00\x00\x00\x00\x00\x00"]]

18446744073709551615, 18446744073709551614, 18446744073709551613, 18446744073709551612, 18446744073709551611, 18446744073709551610, 18446744073709551609, 18446744073709551608, 18446744073709551607, 18446744073709551606, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0

# box.select(["a"], {:index=>5})
[["a", "\n\x00\x00\x00\x00\x00\x00\x00"], ["a", "\t\x00\x00\x00\x00\x00\x00\x00"], ["a", "\b\x00\x00\x00\x00\x00\x00\x00"], ["a", "\a\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x06\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x05\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x04\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x03\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x02\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x00\x00\x00\x00\x00\x00\x00\x00"], ["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"]]

10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, -2, -3, -4, -5, -6, -7, -8, -9, -10

# box.select(["a"], {:index=>6})
[["a", "\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFE\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFD\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFC\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFB\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xFA\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF9\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF8\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF7\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\xF6\xFF\xFF\xFF\xFF\xFF\xFF\xFF"], ["a", "\n\x00\x00\x00\x00\x00\x00\x00"], ["a", "\t\x00\x00\x00\x00\x00\x00\x00"], ["a", "\b\x00\x00\x00\x00\x00\x00\x00"], ["a", "\a\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x06\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x05\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x04\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x03\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x02\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x01\x00\x00\x00\x00\x00\x00\x00"], ["a", "\x00\x00\x00\x00\x00\x00\x00\x00"]]

18446744073709551615, 18446744073709551614, 18446744073709551613, 18446744073709551612, 18446744073709551611, 18446744073709551610, 18446744073709551609, 18446744073709551608, 18446744073709551607, 18446744073709551606, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0

